esp/src/src-react/components/forms/RenameFile.tsx

```diff
@@ -47,21 +54,53 @@ export const RenameFile: React.FunctionComponent<RenameFileProps> = ({
+                                return;
+                            }
+                            const dfuWUResponse = await service.GetDFUWorkunit({ wuid });
+                            const { ID, StateMessage = "", SummaryMessage = "" } = { ...dfuWUResponse?.result };
+                            if (StateMessage === FilesprayStates[5]) { // failed
+                                logger.error(SummaryMessage ?? "");
+                                if (ID) {
+                                    pushUrl(`/dfuworkunits/${ID}`);
+                                }
+                            }
+                            if (ID) {
+                                if (window.location.hash.match(/#\/files\//) === null) {
+                                    pushUrl(`/dfuworkunits/${ID}`);
+                                } else {
+                                    replaceUrl(`/dfuworkunits/${ID}`);
+                                }
                             }
                         });
                     } else {
+                        const requests = [];
+                        const dfuRequests = [];
                         logicalFiles.forEach((logicalFile, idx) => {
-                            const request = { ...data, srcname: logicalFile, dstname: data.targetRenameFile[idx].name };
-                            const requests = [];
-                            requests.push(FileSpray.Rename({ request: request }));
-                            Promise.all(requests).then(_ => {
+                            const request = { ...data, srcname: logicalFile, dstname: data.targetRenameFile[idx].name, DFUServerQueue: "" };
+                            requests.push(service.Rename(request));
+                        });
+                        Promise.all(requests).then(responses => {
+                            responses.forEach(response => {
+                                const wuid = response?.wuid ?? null;
+                                if (wuid) {
+                                    dfuRequests.push(service.GetDFUWorkunit({ wuid }));
+                                }
+                            });
+                            Promise.all(dfuRequests).then(responses => {
+                                responses.forEach(response => {
+                                    const StateMessage = response?.result?.StateMessage ?? FilesprayStates[999];
+                                    const ID = response?.result?.ID;
+                                    if (StateMessage === FilesprayStates[5]) { // failed
+                                        logger.error(response?.result?.SummaryMessage ?? "");
+                                    } else if (ID) {
+                                        window.setTimeout(function () { window.open(`#/dfuworkunits/${ID}`); }, 0);
+                                    }
+                                });
```

Comment:
This code would work equaly well for the case of `logicalFiles?.length===1` - except the bit at the end where you open a new tab or "navigate to"?